go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/containers/buildah
 
-go 1.20
+go 1.21
+
+toolchain go1.21.9
```

Comment:
@nalind I am not sure but I think this directive is a blocking thing because of `podman`, I created a thread here. https://github.com/containers/buildah/pull/5472#issuecomment-2050091140